File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.py
def solution(myString, pat):
    # 방법 2
    # replace 사용. pat을 A -> C, B -> A, C -> B 로 바꾼 뒤, myString에 포함되는지 확인한다.
    return int(pat.replace('A', 'C').replace('B', 'A').replace('C', 'B') in myString)

    # # 방법 1
    # # pat을 A는 B로, B는 A로 바꾼 후, myString에 포함되는지 확인한다.
    # return int(''.join(['B' if ch == 'A' else 'A' for ch in pat]) in myString)

File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.js
function solution(myString, pat) {
    // 방법 2 - map 사용
    // pat를 A -> B, B -> A 로 바꾸고, myString에 포함되는지 확인한다.
    return myString.includes([...pat].map(el => el === 'A' ? 'B' : 'A').join('')) ? 1 : 0
    
    // // 방법 1 - replaceAll 사용
    // // pat을 A -> C, B -> A, C -> B 로 바꾼 뒤, myString에 포함되는지 확인한다.
    // return myString.includes(pat.replaceAll('A', 'C').replaceAll('B', 'A').replaceAll('C', 'B')) ? 1 : 0
}